webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactRepository.scala

```diff
@@ -576,30 +603,80 @@ class CoreNodeFactRepository(
     }
   }
 
-  def save(
+  /*
+   * Save can be used in two fashion: either for setting tenants
+   * or for save. In the second case, we must ensure tenants don't change.
+   * In the first case, we use the existing coreNodeFact for save.
+   */
+  private def internalSave(
+      node:      Either[NodeFact, (NodeId, Option[SecurityTag])]
```

Comment:
For now an either works, but if we need another kind of change we may switch to a type "NodeFactChange", that has several case (ModifyFact, ModifyTenant, XXX ) but if there is only two cases, an either is fine